gcs/cloudStorageClient.go

```diff
@@ -0,0 +1,186 @@
+package gcs
+
+import (
+	"compress/gzip"
+	"context"
+	"encoding/json"
+	"fmt"
+	"path"
+
+	"cloud.google.com/go/storage"
+	"github.com/estafette/estafette-ci-api/config"
+	contracts "github.com/estafette/estafette-ci-contracts"
+)
+
+// CloudStorageClient is the interface for connecting to google cloud storage
+type CloudStorageClient interface {
+	InsertBuildLog(ctx context.Context, buildLog contracts.BuildLog) (err error)
+	InsertReleaseLog(ctx context.Context, releaseLog contracts.ReleaseLog) (err error)
+	GetPipelineBuildLogs(ctx context.Context, buildLog contracts.BuildLog) (updatedBuildLog contracts.BuildLog, err error)
+	GetPipelineReleaseLogs(ctx context.Context, releaseLog contracts.ReleaseLog) (updatedReleaseLog contracts.ReleaseLog, err error)
```

Comment:
Just want to confirm, we're not using these functions yet right?

Reply:
Yes, with the `logReader` set to `database` they won't get used. I'll add the log to start using them depending on that setting already though.